### DISCUSSION TITLE
SESv2 `SendEmail` exceptions

### DISCUSSION THREAD

**Discussion by lonix1** [aws/aws-sdk-net#3523] [Q&A] [answered]
I've scoured the docs and StackOverflow but couldn't find this info.

The SESv2 [`SendEmailAsync`](https://docs.aws.amazon.com/sdkfornet/v3/apidocs/items/SimpleEmailV2/MSimpleEmailServiceV2SendEmailAsyncSendEmailRequestCancellationToken.html) method throws two exceptions which seemingly overlap:
- [BadRequestException](https://docs.aws.amazon.com/sdkfornet/v3/apidocs/items/SimpleEmailV2/TBadRequestException.html) - "The input you provided is invalid. "
- [MessageRejectedException](https://docs.aws.amazon.com/sdkfornet/v3/apidocs/items/SimpleEmailV2/TMessageRejectedException.html) - "The message can't be sent because it contains invalid content. "

These seem very similar; how do they differ?

And:
- [NotFoundException](https://docs.aws.amazon.com/sdkfornet/v3/apidocs/items/SimpleEmailV2/TNotFoundException.html) - "The resource you attempted to access doesn't exist."

What resource is being referred to here?

**Comment by normj** [comment, marked as answer]
The `BadRequestException` if the properties for the request are invalid like there being no `ToAddresses` or `FromAddress`. The `MessageRejectedException` would be in the content going inside the email. That would be something invalid in the raw `Body` or in the `Simple` that makes it an invalid email content. I'm not an expert in SES to know what that would be but the `MessageRejectedException` is specifically the content of the email makes reject.

**Reply by lonix1** [reply]
Thanks for the clarification!

What about `NotFoundException` - I don't understand what "resource" is being referred to?

**Reply by normj** [reply]
My assumption would be if the name of the `ConfigurationSetName` does not exist.

**Reply by lonix1** [reply]
Thanks!

BTW, it seems all the SDKs use the same underlying docs - so in which repo can I submit a ticket to update the docs? I'm sure this specific API is a little confusing to other people too.
